cartographer_ros/cartographer_ros/node.cc

```diff
@@ -459,7 +459,10 @@ void Node::FinishAllTrajectories() {
 
 void Node::FinishTrajectory(const int trajectory_id) {
   carto::common::MutexLocker lock(&mutex_);
+  CHECK(is_active_trajectory_.count(trajectory_id) > 0 &&
```

Comment:
The first check is redundant. You'd also need to rebase once more before we can merge.